Marlin/src/feature/dac/dac_mcp4728.cpp

```diff
@@ -123,8 +123,8 @@ uint8_t MCP4728::getDrvPct(const uint8_t channel) { return uint8_t(100.0 * dac_v
  * Receives all Drive strengths as 0-100 percent values, updates
  * DAC Values array and calls fastwrite to update the DAC.
  */
-void MCP4728::setDrvPct(xyze_uint8_t &pct) {
-  dac_values *= pct.asFloat() * 0.01f * (DAC_STEPPER_MAX);
+void MCP4728::setDrvPct(xyze_uint_t &pct) {
```

Comment:
As described in the related issue, there were 4 different solutions:
- Avoid using structs operators and replace them by `LOOP_XYZE`
- Align types (e.g. declare `pct` as `xyze_uint_t`)
- Add a `asUnsignedInt()` method in structs defined in _types.h_
- Add an assignment operator with a `XYZEval<float>` or `XYZEval<uint8_t>`  typed argument in structs defined in _types.h_

I chose to change the pct type.